### PR TITLE
Authentication fixes for the SDK

### LINKED ISSUE #19344: SDK 11: It's impossible to include credentials with AuthenticationClient methods
### Describe the Bug

When calling for example `refresh()`-method of `AuthenticationClient` module, the request does not include cookies in the headers and there is no way to configure AuthenticationClient methods to include cookies in the headers of the requests.

### To Reproduce

Try calling refresh() method with JS SDK 11 and check out the sent headers. There is no cookies sent with the request.

### Directus Version

v10.1.1

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by br41nslug**
- Allow for cross-origin cookies to be sent with `fetch` (fixes #19344 )
- Fixed non-logical parameter order for refresh request
- Moved helpers to their own folder and renamed `asSearch` to `withSearch` for consistency.

Same-origin cookies:
- This is what `fetch` does out of the box

Cross-origin cookies:
- You'll want to set `credentials: 'include'` on the appropriate composables to send cookies with all requests on that composable
  ```ts
  const client = createDirectus(URL)
    .with(authentication('json', { autoRefresh: false, credentials: 'include' }))
    .with(graphql({ credentials: 'include' }))
    .with(rest({ credentials: 'include' }))
  ```
- Or on specific REST requests using the `withOptions` helper
  ```ts
  const result = await client.request(withOptions(
    readItems('some-collection'),
    { credentials: 'include' }
  ))
  ```

Note: Make sure to double check your CORS and CSP settings when working with cross-origin cookies.

### CODE REVIEW

**Comment by javiercornejo** [comment]
Hi thanks for the contribution. I am having same problem, I think the issue with this implementation is it will only work with rest() but should also work with graphql(), actually I think should work just including authentication(). 

Anyway I am going to try this PR locally and I will comments if I find any.

**Comment by Jon1VK** [comment]
As @javiercornejo said, the https://github.com/directus/directus/issues/19344 issue states only the refresh() function of the authentication() composable, but the problem exists also in the graphql composable and other combosable functions that make fetch requests. It would be great if one could modify all the the requests of all composables when creating the directus client. For example like this. And this would guarantee, that the credentials are sent on every fetch call.

```ts
export const directusClient = createDirectus<DirectusSchema>(DIRECTUS_URL, {
    onRequest: r => {
      r.credentials = 'include';
      return r;
    },
  })
  .with(staticToken(DIRECTUS_API_TOKEN))
  .with(rest());
```

**Comment by br41nslug** [comment]
> Hi @br41nslug thanks for the contribution. I am having same problem, I think the issue with this implementation is it will only work with rest() but should also work with graphql(), actually I think should work just including authentication().
> 
> Anyway I am going to try this PR locally and I will comments if I find any.

You're very right this does not explicitly apply to graphql as i set it up but should work on the `authentication()` handler. Either way glad i left it in draft before my vacation 😄

File: sdk/src/auth/composable.ts [hunk 1]
```diff
@@ -18,19 +19,12 @@ const defaultConfigValues = {
...
 		let refreshTimeout: NodeJS.Timer | null = null;
-		const storage = config.storage ?? memoryStorage();
-
-		const autoRefresh = 'autoRefresh' in config ? config.autoRefresh : defaultConfigValues.autoRefresh;
-
-		const msRefreshBeforeExpires =
-			typeof config.msRefreshBeforeExpires === 'number'
-				? config.msRefreshBeforeExpires
-				: defaultConfigValues.msRefreshBeforeExpires;
+		const storage = authConfig.storage ?? memoryStorage();
```
Comment (paescuj) [inline sdk/src/auth/composable.ts:27]: Couldn't this be in `defaultConfigValues` as well?

**Review by paescuj — APPROVED** [review]
Looks good ~~, besides the little comment~~ 👍

File: sdk/src/auth/composable.ts (hunk 1 above)
Comment (br41nslug) [reply sdk/src/auth/composable.ts:27]: how do you mean? the `defaultConfigValues` are used to build the `authConfig` (similarly to `restConfig` and `gqlConfig`)

Comment (paescuj) [reply sdk/src/auth/composable.ts:27]: Oh, right 😺 Little oversight from my side!

File: sdk/src/auth/types.ts [hunk 2]
```diff
@@ -15,13 +15,14 @@ export interface AuthenticationStorage {
...
-	msRefreshBeforeExpires?: number;
+	autoRefresh: boolean;
+	msRefreshBeforeExpires: number;
+	credentials: RequestCredentials;
 	storage?: AuthenticationStorage;
 }
 
 export interface AuthenticationClient<_Schema extends object> {
-	login(email: string, password: string, options: loginOptions): Promise<AuthenticationData>;
+	login(email: string, password: string, options?: loginOptions): Promise<AuthenticationData>;
```
Comment (paescuj) [inline sdk/src/auth/types.ts:25]: A bit unrelated and nitpicking, but just realized the type `loginOptions` starts with lower case letter. Is there a specific reason for this or can we change it to a capital letter to align with all other types?

Comment (rijkvanzanten) [reply sdk/src/auth/types.ts:25]: Great find

File: .changeset/rude-games-decide.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"docs": patch
+"@directus/sdk": major
+---
+
+Allow for cross-origin cookies to be sent with fetch, Fixed parameter order for refresh request and moved helpers to their own folder and renamed asSearch to withSearch for consistency
```
[the remark this answers is not in the document]
Comment (br41nslug) [reply .changeset/rude-games-decide.md:6]: > moved helpers to their own folder, Isn't affecting users as the import is still the same, right?

Correct nothing changes there for the user.

> renamed asSearch to withSearch for consistency

This is definitely still a breaking change

> It seems like the withOptions is newly introduced

Correct that is a new helper i introduced for easier options editing.

**Review by azrikahar — COMMENTED** [review]
PR LGTM! Just a quick question regarding the `breaking-changes.md`, do we need to mention/include the renaming of `asSearch` to `withSearch` as well?

**Comment by br41nslug** [comment]
> PR LGTM! Just a quick question regarding the `breaking-changes.md`, do we need to mention/include the renaming of `asSearch` to `withSearch` as well?

Yes we do, nice catch

File: docs/releases/breaking-changes.md [hunk 4]
```diff
@@ -11,6 +11,39 @@ these to a minimum, but rest assured we only make them with good reason.
 
 Starting with Directus 10.0, here is a list of potential breaking changes with remedial action you may need to take.
 
+## Version 10.6.2
+
+### Swapping Parameters and Auth Mode for Refresh Method in the SDK
```
Comment (paescuj) [inline docs/releases/breaking-changes.md:16]: Oversight from my side: We should use past tense as it's the case with the existing titles
```suggestion
### Swapped Parameters and Auth Mode for Refresh Method in the SDK
```

File: docs/releases/breaking-changes.md [hunk 5]
```diff
@@ -11,6 +11,39 @@ these to a minimum, but rest assured we only make them with good reason.
...
+const result = await client.request(refresh());
+const result = await client.request(refresh('cookie'));
+
+// refresh http request using json
+const result = await client.request(refresh('json', refresh_token));
+```
+
+:::
+
+### Renaming helper functions in the SDK
```
Comment (paescuj) [inline docs/releases/breaking-changes.md:43]:
```suggestion
### Renamed Helper Function in the SDK
```

**Pull request merged**